### PR TITLE
Introduce @nordeck/element-web-widget-toggles-module

### PR DESCRIPTION
**Pull request by charlynguyen**
**:heavy_check_mark: Checklist**

- [x] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [x] Added or updated documentation.
- [x] Tests for new functionality and regression tests for bug fixes.
- [x] Screenshots or videos attached (for UI changes).
- [x] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

**Comment by charlynguyen** [comment]
> I've followed the README instructions to set up the widget toggles locally. It looks like that the toggles are always „one room behind“. Here is a recording:
> 
> toggles [ toggles ](https://private-user-images.githubusercontent.com/6216686/304374938-3baf2ddb-b5e0-4365-8553-14d3eedd528a.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MjA1NDksIm5iZiI6MTcwNzgyMDI0OSwicGF0aCI6Ii82MjE2Njg2LzMwNDM3NDkzOC0zYmFmMmRkYi1iNWUwLTQzNjUtODU1My0xNGQzZWVkZDUyOGEuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTNUMTAzMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWU3MDNjZWM1Y2FkYjA0YzgwZmIwYjFiODFlMmIwMTM0ZDhhZGMxY2NkZWNhNzZjZTkzNDE4MDk2YjRkZjVjNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ukSb01NasXd756LHJcEF18UOTbUNTs843nFNyLaC7UA) [ ](https://private-user-images.githubusercontent.com/6216686/304374938-3baf2ddb-b5e0-4365-8553-14d3eedd528a.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MjA1NDksIm5iZiI6MTcwNzgyMDI0OSwicGF0aCI6Ii82MjE2Njg2LzMwNDM3NDkzOC0zYmFmMmRkYi1iNWUwLTQzNjUtODU1My0xNGQzZWVkZDUyOGEuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTNUMTAzMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWU3MDNjZWM1Y2FkYjA0YzgwZmIwYjFiODFlMmIwMTM0ZDhhZGMxY2NkZWNhNzZjZTkzNDE4MDk2YjRkZjVjNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ukSb01NasXd756LHJcEF18UOTbUNTs843nFNyLaC7UA)
> 
> Can you also re-test it @charlynguyen ? If I am doing something wrong we can also have a remote session to figure the issue out.

You are not doing anything wrong. We have an issue since a certain change in Element Web.

**Review by weeman1337 — DISMISSED** [review]
Looks good code-wise :+1: 

Unfortunately I am unable to test it because of the known issue. There is one small question about the dependencies.

**Comment by weeman1337** [comment]
> > I've followed the README instructions to set up the widget toggles locally. It looks like that the toggles are always „one room behind“. Here is a recording:
> > toggles
> >       
> >         [
> >           
> >             toggles
> >           
> >         ](https://private-user-images.githubusercontent.com/6216686/304374938-3baf2ddb-b5e0-4365-8553-14d3eedd528a.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MjA1NDksIm5iZiI6MTcwNzgyMDI0OSwicGF0aCI6Ii82MjE2Njg2LzMwNDM3NDkzOC0zYmFmMmRkYi1iNWUwLTQzNjUtODU1My0xNGQzZWVkZDUyOGEuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTNUMTAzMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWU3MDNjZWM1Y2FkYjA0YzgwZmIwYjFiODFlMmIwMTM0ZDhhZGMxY2NkZWNhNzZjZTkzNDE4MDk2YjRkZjVjNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ukSb01NasXd756LHJcEF18UOTbUNTs843nFNyLaC7UA)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/6216686/304374938-3baf2ddb-b5e0-4365-8553-14d3eedd528a.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MjA1NDksIm5iZiI6MTcwNzgyMDI0OSwicGF0aCI6Ii82MjE2Njg2LzMwNDM3NDkzOC0zYmFmMmRkYi1iNWUwLTQzNjUtODU1My0xNGQzZWVkZDUyOGEuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTNUMTAzMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWU3MDNjZWM1Y2FkYjA0YzgwZmIwYjFiODFlMmIwMTM0ZDhhZGMxY2NkZWNhNzZjZTkzNDE4MDk2YjRkZjVjNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ukSb01NasXd756LHJcEF18UOTbUNTs843nFNyLaC7UA)
> >         
> >       
> > Can you also re-test it @charlynguyen ? If I am doing something wrong we can also have a remote session to figure the issue out.
> 
> You are not doing anything wrong. We have an open issue since a certain change in Element Web.

Has an issue already been opened in Element Web for this problem?

File: packages/element-web-widget-toggles-module/package.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "@matrix-org/react-sdk-module-api": "^2.3.0",
+    "joi": "^17.11.0",
+    "styled-components": "^6.1.8"
+  },
+  "peerDependencies": {
+    "react": "^17",
+    "react-dom": "^17"
+  },
+  "resolutions": {
+    "styled-components": "^6"
```
[the remark this answers is not in the document]
Comment (charlynguyen) [reply packages/element-web-widget-toggles-module/package.json:42]: It might not be required, but highly recommended to be safe.

**Comment by charlynguyen** [comment]
> Has an issue already been opened in Element Web for this problem?

There is no issue yet. Meanwhile it is suggested by Element
- to listen to the `Action.ViewRoom` or
- introduce a dedicated event for the widgets load.

**Pull request merged**